doc/METRICS.md

```diff
@@ -0,0 +1,36 @@
+# Bandwidth Metrics
+
+* Metrics gathering should be optional, as there is a performance hit to using it.
+* Metrics do NOT currently contain OS level stats, only libp2p application level metrics. For example, TCP messages (ACK, FIN, etc) are not accounted for.
+
+## Tracking
+* When a transport hands off a connection for upgrading, metrics are hooked up if turned on.
+* When a stream is created, metrics will be tracked on that stream and associated to that streams protocol.
+
+### Questions?
+* Should we track crypto protocol bandwidth for crypto handshaking? We won't know the remote peer id for sure until crypto is done, so we will need some way to lazily track by id.
+* Should we track muxer protocol bandwidth for muxer selection? This is likely not worth it.
```

Comment:
I think it would be nice to track the crypto negotiation. We can postpone a decision for this for when we iterate on `connMgr`.

Same for `muxer`, I don't think it is relevant either, but we can revisit

Reply:
I think muxer selection won't end up being needed. We should be able to extrapolate this information by getting the stats for a peer and then checking the muxer associated with their connection(s).

I'll start without crypto handshake tracking for now, and then add that in once the base is done.